### PR TITLE
Allow users to see Cost of Living Hub on integration

### PR DESCRIPTION
**Pull request by chao-xian**
## What

Sets it so that users can view the Cost of living page on integration while being hidden behind a feature flag on Staging and Production.

Previously this was only hidden on all environments except Heroku and Dev (we can set the equivalent env var for Heroku previews in the Heroku settings for the app).

## Why

This will allow us to test things more accurately.

[Trello](https://trello.com/c/0rJHy0zc/1236-page-is-on-production-and-hidden-viewable-only-on-integration-m)

[Related PR in puppet](https://github.com/alphagov/govuk-puppet/pull/11781)

[Related PR in special-route-publisher](https://github.com/alphagov/special-route-publisher/pull/224)

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

### CODE REVIEW

File: config/environments/production.rb [hunk 1]
```diff
@@ -83,7 +83,7 @@
 
   # Configuration to stop Cost of Living landing page showing in production before
   # campaign goes live
-  if (ENV["BASIC_AUTH_USERNAME"] && ENV["BASIC_AUTH_PASSWORD"] && ENV["HEROKU_APP_NAME"]) || ENV["JENKINS"] || ENV["PUBLISHING_E2E_TESTS_COMMAND"]
+  if (ENV["BASIC_AUTH_USERNAME"] && ENV["BASIC_AUTH_PASSWORD"] && ENV["HEROKU_APP_NAME"]) || (ENV["BASIC_AUTH_USERNAME"] && ENV["BASIC_AUTH_PASSWORD"] && ENV["ENVIRONMENT"] == "integration") || ENV["JENKINS"] || ENV["PUBLISHING_E2E_TESTS_COMMAND"]
```
Comment (leenagupte) [inline config/environments/production.rb:86]: Have you considered setting an environment variable in the environment to show the page?
e.g. https://github.com/alphagov/govuk-puppet/blob/98d30fd7b7dc3e9c3c50e178c3aa57c24a5cf3dc/hieradata_aws/integration.yaml#L36

You could leave `unreleased_features_enabled` set to true in integration permanently so your changes would always show up there.

We've done something similar in collections publisher: https://github.com/alphagov/collections-publisher/search?q=unreleased+features

Comment (chao-xian) [reply config/environments/production.rb:86]: Nice! I'll rework - thanks!

Comment (chao-xian) [reply config/environments/production.rb]: @leenagupte have reworked this PR now. Thanks!

**Review by leenagupte — COMMENTED** [review]
This is looking good. There are couple of minor inline comments.

File: config/environments/development.rb [hunk 2]
```diff
@@ -61,7 +61,7 @@
 
   # Configuration to stop Cost of Living landing page showing in production before
   # campaign goes live
-  config.unreleased_features = true
+  config.unreleased_features = false
```
[the remark this answers is not in the document]
Comment (chao-xian) [reply config/environments/development.rb]: Argh that was a testing situation forgot to undo error.

File: config/environments/production.rb [hunk 3]
```diff
@@ -82,10 +82,8 @@
   # config.logger = ActiveSupport::TaggedLogging.new(Syslog::Logger.new "app-name")
 
   # Configuration to stop Cost of Living landing page showing in production before
-  # campaign goes live
-  if (ENV["BASIC_AUTH_USERNAME"] && ENV["BASIC_AUTH_PASSWORD"] && ENV["HEROKU_APP_NAME"]) || (ENV["BASIC_AUTH_USERNAME"] && ENV["BASIC_AUTH_PASSWORD"] && ENV["ENVIRONMENT"] == "integration") || ENV["JENKINS"] || ENV["PUBLISHING_E2E_TESTS_COMMAND"]
-    config.unreleased_features = true
-  end
+  # page goes live. Puppet has this set to true: https://github.com/alphagov/govuk-puppet/pull/11781
```
[the remark this answers is not in the document]
Comment (chao-xian) [reply config/environments/production.rb]: Comment updated.

File: config/environments/test.rb [hunk 4]
```diff
@@ -54,7 +54,7 @@
 
   # Configuration to stop Cost of Living landing page showing in production before
   # campaign goes live
-  config.unreleased_features = true
+  config.unreleased_features = false
```
[the remark this answers is not in the document]
Comment (chao-xian) [reply config/environments/test.rb:57]: Remove the line in the spec helper. Forgot it was in there.

File: app/controllers/cost_of_living_landing_page_controller.rb [hunk 5]
```diff
@@ -1,13 +1,17 @@
 class CostOfLivingLandingPageController < ApplicationController
   slimmer_template "gem_layout_full_width"
-  
+
   def show
-    @content_item = { "locale" => "en" }
+    if Rails.application.config.unreleased_features
+      @content_item = { "locale" => "en" }
```
[the remark this answers is not in the document]
Comment (chao-xian) [reply app/controllers/cost_of_living_landing_page_controller.rb]: Ah good spot! Have done as suggested. Thank you!

File: config/environments/test.rb (hunk 4 above)
Comment (leenagupte) [inline config/environments/test.rb:57]: This is not a no blocker. However, I would set this value to `true`. If you do that you'll have less clean to do in the tests when this page goes live.

**Review by leenagupte — APPROVED** [review]
👍 Looks good.  A minor comment, but no blockers.

**Pull request merged**